ksql-parser/src/test/java/io/confluent/ksql/parser/json/LogicalSchemaDeserializerTest.java

```diff
@@ -83,10 +86,43 @@ public void shouldDeserializeSchemaWithKeyAfterValue() throws Exception {
         .build()));
   }
 
+  @Test
+  public void shouldDeserializeSchemaWithSource() throws IOException {
+    // Given:
+    final String json = "\"SOURCE.K0 BIGINT KEY, SOURCE.V0 STRING\"";
+
+    // When:
+    final LogicalSchema schema = MAPPER.readValue(json, LogicalSchema.class);
+
+
+    Assert.assertThat(schema, is(LogicalSchema.builder()
+        .noImplicitColumns()
+        .keyColumn(SourceName.of("SOURCE"), ColumnName.of("K0"), SqlTypes.BIGINT)
+        .valueColumn(SourceName.of("SOURCE"), ColumnName.of("V0"), SqlTypes.STRING)
+        .build()));
+  }
+
+  @Test
+  public void shouldAddImplicitColumns() throws IOException {
+    // Given:
+    final ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new TestModule(true));
+    final String json = "\"K0 BIGINT KEY, V0 STRING\"";
+
+    // When:
+    final LogicalSchema schema = mapper.readValue(json, LogicalSchema.class);
+
+
+    Assert.assertThat(schema, is(LogicalSchema.builder()
```

Comment:
I think this test would be more robust if it actually explicitly checked for the presence of the meta columns, e.g. 

```java
assertThat(schema.meta(), is(not(empty()));
```

It's also more readable as to what the test is testing.
```